solr/test-framework/src/java/org/apache/solr/cloud/SolrCloudTestCase.java

```diff
@@ -385,4 +389,39 @@ public static Map<String, String> mapReplicasToReplicaType(DocCollection collect
     }
     return replicaTypeMap;
   }
+
+  protected void setSliceState(String collection, String slice, Slice.State state)
```

Comment:
SolrCloudTestCase is supposed to be rather simple.  The method here is rather exotic.  Maybe there's a SliceMutator or some other place where it might better fit as a static method perhaps?  Just an idea.
Just checking -- you moved this from DeleteShardTest?  And made no modifications?

Reply:
I made no modifications to the method, but I have moved it now to it's own file so that it can be reused more easily. I agree that SolrCloudTestCase really wasn't the right place for it.